Formula/p/pyside@2.rb

```diff
@@ -91,7 +100,6 @@ def install
     system "cmake", "-S", ".", "-B", "build",
                     "-DPYTHON_EXECUTABLE=#{which(python3)}",
                     "-DCMAKE_INSTALL_RPATH=#{rpaths.join(";")}",
-                    "-DFORCE_LIMITED_API=yes",
```

Comment:
This is an ABI break which we avoid doing in versioned formulae. The soname has now changed from `abi3` to `cpython-312-darwin`.

Reply:
Deprecating in #193877

Reply:
> That policy bullet (specifically the latter parts on revision bumps) would also imply that versioned formulae with Python bindings will be locked to a specific Python since the instant we update the Python version it will break dependents and require revision bumps

The idea around the limtied API here was to cover this. The dylib should be forward compatible so we compile with Python 3.10 and it would work (with symlinks potentially) with 3.11+ too.

(This is all assuming upstream correctly implemented that of course.)

You can see on PyPI how they have one precompiled wheel for every 3.x: https://pypi.org/project/PySide2/#files

Reply:
> The dylib should be forward compatible so we compile with Python 3.10 and it would work (with symlinks potentially) with 3.11+ too.

I previously tried using Python 3.12 with the dylib and it segfaulted, but didn't care enough to debug further.

Reply:
> Also note that the Limited API is not necessarily stable: compiling with `Py_LIMITED_API` with Python 3.8 means that the extension will run with Python 3.12, but it will not necessarily compile with Python 3.12. 

More info at: https://docs.python.org/3/c-api/stable.html

Reply:
> I previously tried using Python 3.12 with the dylib and it segfaulted, but didn't care enough to debug further.

Fair, probably a bug then. But not worth finding which commit fixes it.